vertx-pojo-mapper-common/src/main/java/de/braintags/vertx/jomnigate/dataaccess/query/IIndexedField.java

```diff
@@ -51,4 +47,22 @@ static IIndexedField getIndexedField(String name, Class<?> pojoClass)
           + "' must be static, final, and of type " + IIndexedField.class.getName());
   }
 
+  default String getColumnName(IDataStore<?, ?> datastore) {
+    return getColumnName(datastore.getMapperFactory().getMapper(getClass()));
+  }
+
+  default String getColumnName(IMapper<?> mapper) {
+    String fieldName = getFieldName();
+    String subFieldName = "";
+    int i = fieldName.indexOf('.');
+    if (i > 0) {
+      subFieldName = fieldName.substring(i);
+      fieldName = fieldName.substring(0, i);
+    }
+    IProperty field = mapper.getField(fieldName);
+    if (field == null)
+      throw new de.braintags.vertx.jomnigate.exception.NoSuchFieldException(getFieldName());
```

Comment:
I think it makes sense, the java one is thrown if the IndexedField is missing in the java class, and the jomnigate one is thrown when a field could not be found inside the mapper.